rclcpp/test/benchmark/benchmark_node.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+class NodePerformanceTest : public PerformanceTest
+{
+public:
+  void SetUp(::benchmark::State & state)
+  {
+    rclcpp::init(0, nullptr);
+    performance_test_fixture::PerformanceTest::SetUp(state);
+  }
+
+  void TearDown(::benchmark::State & state)
+  {
+    rclcpp::shutdown();
+    performance_test_fixture::PerformanceTest::TearDown(state);
+  }
+};
+
+BENCHMARK_F(NodePerformanceTest, create_node)(benchmark::State & state)
+{
+  std::vector<std::unique_ptr<rclcpp::Node>> nodes(10000);
+  for (auto _ : state) {
+    auto node = std::make_unique<rclcpp::Node>("node");
+    benchmark::DoNotOptimize(node);
+    benchmark::ClobberMemory();
```

Comment:
I could never observe any change in behavior when I used these. Could you share the results with/without it?

Reply:
The difficult thing about comparing results with this is that `DoNotOptimize()` doesn't remove all optimizations. So sometimes it's important to include and sometimes it makes no difference. Because ultimately the compiler's optimizations will be different depending on the platform and optimization level, I thought it would be better to include it. Even though for this example it doesn't make much difference.

Here are two simple examples which shows what gets optimized and what doesn't.
https://godbolt.org/z/c6aP3d  with DoNotOptimize
https://godbolt.org/z/8ee5sT without

A little more context:
https://github.com/google/benchmark/issues/242#issuecomment-230056319